Remove fake tensor cache clearing in dynamo

eellison opened the pull request:
Summary: We originally cleared the cache of the converter to avoid memory leaks; now that the cache uses a weak map this is no longer necessary. Clearing of the cache caused an error in an interaction with the minifier because the minifier uses delayed compilation, so the cleanup had occurred before inductor was invoked.

Test Plan: Memory regression is being checked via dashboard and on master.

Differential Revision: D42858624



cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

ezyang reviewed (approved):
don't forget to check memory compression on benchmark suite

eellison committed:
Remove fake tensor cache clearing in dynamo (#93304)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/93304

We originally cleared the cache of the converter to avoid memory leaks; now that the cache uses a weak map this is no longer necessary. Clearing of the cache caused an error in an interaction with the minifier because the minifier uses delayed compilation, so the cleanup had occurred before inductor was invoked.

Test Plan: Memory regression is being checked via dashboard and on master.

Reviewed By: brad-mengchi

Differential Revision: D42858624

fbshipit-source-id: bfa653ecd345b79f3dd182e11a426f5cfe2cce0a

eellison commented:
@pytorchbot merge -f "Flakey failure"
